includes/components/class-attribute.php

```diff
@@ -1860,4 +1863,30 @@ public function disable_jetpack_search( $enabled, $query ) {
 
 		return $enabled;
 	}
+
+	/**
+	 * Redirect archive page.
+	 */
+	public function redirect_archive_page() {
+		if ( is_post_type_archive( hp\prefix( $this->get_models() ) ) ) {
+
+			// Get archive post name.
+			$post_type_name = get_post_type();
+
+			if ( ! $post_type_name ) {
+				$post_type_name = hp\get_array_value( (array) get_queried_object(), 'name', null );
+			}
+
+			// Get page ID.
+			$page_id = absint( get_option( hp\prefix( 'page_' . hp\unprefix( $post_type_name ) . 's' ) ) );
+
+			if ( ! $page_id ) {
```

Comment:
it's strange that you use a guard clause here but the whole function is wrapped with an if:)